File: Medium/0167_two-sum-ii-input-array-is-sorted/Solution.py
class Solution:
    def twoSum(self, numbers: list[int], target: int) -> list[int]:
        first = 0
        last = len(numbers) - 1
        while first < last:
            sum_val = numbers[first] + numbers[last]
            if sum_val > target:
                last -= 1
            elif sum_val < target:
                first += 1
            else:
                return [first + 1, last + 1]
        return [-1, -1]

File: Medium/0167_two-sum-ii-input-array-is-sorted/Solution.cpp
#include <vector>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& numbers, int target) {
        int first = 0;
        int last = numbers.size() - 1;
        while (first < last) {
            int sum = numbers[first] + numbers[last];
            if (sum > target) {
                last--;
            } else if (sum < target) {
                first++;
            } else {
                return {first + 1, last + 1};
            }
        }
        return {-1, -1};
    }
};